openprescribing/frontend/management/commands/scrape_numbers.py

```diff
@@ -0,0 +1,243 @@
+        'accounts',
+        'admin',
+        'api',
+        'bnf',
+        'bookmarks',
+        'chemical',
+        'dmd',
+        'docs',
+    ]
+
+    # get_resolver().reverse_dict is a dict that maps view names or view
+    # functions to a data structure that describes how requests should be
+    # dispatched.
+    for k, v in get_resolver().reverse_dict.items():
+
+        # Ignore records where the key is a view function.
+        if not isinstance(k, str):
+            continue
+
+        name = k
+        pattern = v[0][0][0]
+        keys = v[0][0][1]
+
+        # Ignore records starting with prefixes we're not interested in.
+        if any(pattern.startswith(prefix) for prefix in prefixes_to_ignore):
+            continue
+
+        # Ignore any URLs that 
+        if '%' not in pattern and 'all-england' not in pattern:
+            continue
+
+        path = bulid_path(pattern, keys)
+        yield name, path
+
+
+def bulid_path(pattern, keys):
+    '''Replace placeholders in `pattern` to build path to be scraped.
+
+    `pattern` will be a string like "measure/%(measure)s/ccg/%(entity_code)s/".
+    `keys` will be a list like ["measure", "entity_code"].
+    '''
+
+    substitutions = {
+        'practice_code': 'L83100',
+        'ccg_code': '15N',
+        'stp_code': 'E54000037',
+        'regional_team_code': 'Y58',
+        'bnf_code': '0205051R0BBAIAN',
+        'measure': 'ace',
+    }
+
+    path = pattern
+
+    for key in keys:
+        subst_key = None
+
+        if key in ['code', 'entity_code']:
+            for token in ['practice', 'ccg', 'stp', 'regional-team']:
+                if token in pattern:
+                    subst_key = '{}_code'.format(token).replace('-', '_')
+            if subst_key is None:
+                subst_key = 'bnf_code'
+        else:
+            subst_key = key
+
+        path = path.replace(
+            '%({})s'.format(key), 
+            substitutions[subst_key]
+        )
+
+    assert '%' not in path, 'Could not interpolate ' + name
+
+    return path
+
+
+def extract_numbers(browser, path, name, log_path):
+    '''Request URL, write copy of response to log_path, and return list of
+    anything that looks like an interesting number in the response.
+    '''
+
+    url = 'https://openprescribing.net/' + path
+    browser.get(url)
+
+    # Wait until all AJAX requests are complete.
+    while browser.execute_script('return jQuery.active > 0'):
+        sleep(0.1)
+
+    source = browser.page_source
+
+    with open(os.path.join(log_path, name + '.html'), 'w') as f:
+        f.write(source.encode('utf8'))
+
+    doc = BeautifulSoup(source, 'html.parser')
+    body = doc.find('body')
+
+    # The SVG charts contain all sorts of things that look like numbers (eg
+    # rgb(255,255,255)) so we drop all <svg> elements.
+    for tag in body.find_all('svg'):
+        tag.decompose()
+
+    rx = re.compile(u'''
+        £[\d,\.]+           # Anything that looks like a cost
```

Comment:
I found one instance of `&pound;` in the javascript code, too.

Reply:
Bleugh.